python/dgl/contrib/graph_partition_book.py

```diff
@@ -0,0 +1,254 @@
+            assert 'node_map' in self._part_meta, "cannot get the node map."
+            self._nid2partid = np.load(self._part_meta['node_map'])
+
+        return self._nid2partid[nids]
+        
+        
+    def eid2partid(self, eids):
+        """From global edge IDs to partition IDs
+
+        Parameters
+        ----------
+        eids : tensor
+            global edge IDs
+
+        Returns
+        -------
+        tensor
+            partition IDs
+        """
+        if self._eid2partid is None:
+            assert 'edge_map' in self._part_meta, "cannot get the edge map."
+            self._eid2partid = np.load(self._part_meta['edge_map'])
+
+        return self._eid2partid
+    
+
+    def partid2nids(self, partid):
+        """From partition id to node IDs
+
+        Parameters
+        ----------
+        partid : int
+            partition id
+
+        Returns
+        -------
+        tensor
+            node IDs
+        """
+        if len(self._partid2nids) == 0:
+            sorted_nid = F.tensor(np.argsort(F.asnumpy(self._nid2partid)))
+            part, count = np.unique(F.asnumpy(self._nid2partid), return_counts=True)
+            assert len(part) == self._num_partitions
+            start = 0
+            for offset in count:
+                part_nids = sorted_nid[start:start+offset]
+                start += offset
+                self._partid2nids.append(part_nids)
+
+        return self._partid2nids[partid]
+
+
+    def partid2eids(self, partid):
+        """From partition id to edge IDs
+
+        Parameters
+        ----------
+        partid : int
+            partition id
+
+        Returns
+        -------
+        tensor
+            edge IDs
+        """
+        if len(self._partid2eids) == 0:
+            sorted_eid = F.tensor(np.argsort(F.asnumpy(self._eid2partid)))
+            part, count = np.unique(F.asnumpy(self._eid2partid), return_counts=True)
+            assert len(part) == self._num_partitions
+            start = 0
+            for offset in count:
+                part_eids = sorted_eid[start:start+offset]
+                start += offset
+                self._partid2eids.append(part_eids)
+
+        return self._partid2eids[partid]
+
+    
+    def nid2localnid(self, nids, partid):
+        """Get local node IDs within the given partition.
+
+        Parameters
+        ----------
+        nids : tensor
+            global node IDs
+        partid : int
+            partition ID
+
+        Returns
+        -------
+        tensor
+             local node IDs
+        """
+        if len(self._nidg2l) == 0:
+            for partid in range(self._num_partitions):
+                global_id = self._part_graphs[partid].ndata[NID]
+                max_global_id = F.asnumpy(global_id).amax()
+                g2l = F.zeros((max_id+1), F.int64, F.cpu())
+                g2l[global_id] = F.arange(0, len(global_id))
+                self._nidg2l.append(g2l)
+
+        return self._nidg2l[partid]
```

Comment:
where does nids go?

Reply:
It shoud be `return self._nidg2l[partid][nids]`